Shared.IntegrationTests/AdminApi/ManageFoldersTest.cs

```diff
@@ -2,70 +2,249 @@
 using System.Linq;
 using System.Net;
 using System.Threading;
+using System.Threading.Tasks;
 using CloudinaryDotNet.Actions;
 using NUnit.Framework;
 
 namespace CloudinaryDotNet.IntegrationTest.AdminApi
 {
     public class ManageFoldersTest : IntegrationTestBase
     {
-        // For this test to work, "Auto-create folders" should be enabled in the Upload Settings, so this test is disabled by default.
-        [Test, IgnoreFeature("auto_create_folders")]
+        private string _rootFolder1;
+        private string _rootFolder2;
+        private string _subFolder1;
+        private string _subFolder2;
+
+        [Test]
         public void TestFolderApi()
         {
-            // should allow to list folders and subfolders
-            var subFolder1 = $"{m_folderPrefix}1/test_subfolder1";
-            var subFolder2 = $"{m_folderPrefix}1/test_subfolder2";
-
-            var publicIds = new List<string> {
-                $"{m_folderPrefix}1/item",
-                $"{m_folderPrefix}2/item",
-                $"{subFolder1}/item",
-                $"{subFolder2}/item"
+            PrepareFoldersForTests();
+
+            WaitForServerUpdate();
+
+            var result = m_cloudinary.RootFolders();
+            AssertGetRootFolders(result);
+
+            result = m_cloudinary.SubFolders(_rootFolder1);
+
+            AssertGetSubFolders(result);
+
+            result = m_cloudinary.SubFolders(m_folderPrefix);
+
+            AssertGetSubFoldersError(result);
+
+            var deletionRes = m_cloudinary.DeleteFolder($"{_subFolder1}_");
+
+            AssertDeleteFolderError(deletionRes);
+
+            m_cloudinary.DeleteResourcesByPrefix(_subFolder1);
+
+            deletionRes = m_cloudinary.DeleteFolder(_subFolder1);
+
+            AssertDeleteFolder(deletionRes);
+
+            DeleteFoldersForTests();
```

Comment:
Please move this and all `DeleteFoldersForTests()` to `TearDown()`.

Here it is never executed if at least one assertion fails and folders are not deleted.

Reply:
Updated